src/sentry/snuba/events.py

```diff
@@ -39,43 +39,96 @@ class Columns(Enum):
+        "contexts[device.arch]",
+        "contexts[device.arch]",
+        "device.arch",
     )
-    DEVICE_ARCH = Column("events.device_arch", "device_arch", "device_arch", "device.arch")
     DEVICE_BATTERY_LEVEL = Column(
-        "events.device_battery_level",
-        "device_battery_level",
-        "device_battery_level",
+        "events.contexts[device.battery_level]",
+        "contexts[device.battery_level]",
+        "contexts[device.battery_level]",
         "device.battery_level",
     )
-    DEVICE_BRAND = Column("events.device_brand", "device_brand", "device_brand", "device.brand")
+    DEVICE_BRAND = Column(
+        "events.contexts[device.brand]",
+        "contexts[device.brand]",
+        "contexts[device.brand]",
+        "device.brand",
+    )
     DEVICE_CHARGING = Column(
-        "events.device_charging", "device_charging", "device_charging", "device.charging"
+        "events.contexts[device.charging]",
+        "contexts[device.charging]",
+        "contexts[device.charging]",
+        "device.charging",
     )
     DEVICE_LOCALE = Column(
-        "events.device_locale", "device_locale", "device_locale", "device.locale"
+        "events.contexts[device.locale]",
+        "contexts[device.locale]",
+        "contexts[device.locale]",
+        "device.locale",
     )
     DEVICE_MODEL_ID = Column(
-        "events.device_model_id", "device_model_id", "device_model_id", "device.model_id"
+        "events.contexts[device.model_id]",
+        "contexts[device.model_id]",
+        "contexts[device.model_id]",
+        "device.model_id",
+    )
+    DEVICE_NAME = Column(
+        "events.contexts[device.name]",
+        "contexts[device.name]",
+        "contexts[device.name]",
+        "device.name",
     )
-    DEVICE_NAME = Column("events.device_name", "device_name", "device_name", "device.name")
     DEVICE_ONLINE = Column(
-        "events.device_online", "device_online", "device_online", "device.online"
+        "events.contexts[device.online]",
+        "contexts[device.online]",
+        "contexts[device.online]",
+        "device.online",
     )
     DEVICE_ORIENTATION = Column(
-        "events.device_orientation",
-        "device_orientation",
-        "device_orientation",
+        "events.contexts[device.orientation]",
+        "contexts[device.orientation]",
+        "contexts[device.orientation]",
         "device.orientation",
     )
     DEVICE_SIMULATOR = Column(
-        "events.device_simulator", "device_simulator", "device_simulator", "device.simulator"
+        "events.contexts[device.simulator]",
+        "contexts[device.simulator]",
+        "contexts[device.simulator]",
+        "device.simulator",
+    )
+    DEVICE_UUID = Column(
+        "events.contexts[device.uuid]",
+        "contexts[device.uuid]",
+        "contexts[device.uuid]",
+        "device.uuid",
     )
-    DEVICE_UUID = Column("events.device_uuid", "device_uuid", "device_uuid", "device.uuid")
     GEO_COUNTRY_CODE = Column(
```

Comment:
The geo fields could all move to contexts too.

Reply:
They are handled differently than these other fields in snuba, so I left them the same for now. I think it might be better to do a follow up PR with those ones (sdk ones as well).